xen/common/device_tree.c

```diff
@@ -1994,6 +1994,8 @@ static void __init __unflatten_device_tree(const void *fdt,
     *allnextp = NULL;
 
     dt_dprintk(" <- unflatten_device_tree()\n");
+
```

Comment:
It is not your fault, but shouldn't NULL be returned if we failed to allocate memory (line 1979)?